Kc/kedro integration advanced

jakubczakon opened the pull request:
* started working on advanced use cases

jakubczakon commented:
@kamil-kaczmarek seems like the failing ones are about some random connection lost error and not Kedro pipelines at all.

kamil-kaczmarek reviewed (commented):
For the most part the code is created by kerdo -> is that right?

There a re also test directories - was that intentional? Do you plan to run these tests, or are they created by kedro automatically?

jakubczakon commented:
@kamil-kaczmarek yes, the code is mostly Kedro code, the only files where Neptune changes live are:
* integrations-and-supported-tools/kedro/scripts/kedro_neptune_advanced/conf/base/catalog.yml
* integrations-and-supported-tools/kedro/scripts/kedro_neptune_advanced/conf/base/catalog_neptune.yml
* integrations-and-supported-tools/kedro/scripts/kedro_neptune_advanced/conf/base/neptune.yml
* integrations-and-supported-tools/kedro/scripts/kedro_neptune_advanced/conf/base/parameters.yml
* integrations-and-supported-tools/kedro/scripts/kedro_neptune_advanced/conf/local/credentials_neptune.yml
*  integrations-and-supported-tools/kedro/scripts/kedro_neptune_advanced/src/kedro_neptune_advanced/pipelines/data_science/nodes.py
* integrations-and-supported-tools/kedro/scripts/kedro_neptune_advanced/src/kedro_neptune_advanced/pipelines/data_science/pipeline.py
* integrations-and-supported-tools/kedro/scripts/kedro_neptune_advanced/data/01_raw/iris.csv
* integrations-and-supported-tools/kedro/scripts/kedro_neptune_advanced/data/01_raw/iris_v2.csv

I don't plan to run any tests -> they were just included with Kedro.
Actually I'll try and download/run this on clean env without the tests folders just to see if Kedro is fine with it.

jakubczakon commented:
@kamil-kaczmarek now I added all the files from those kedro projects and adjusted the test run_example.sh to make it work. 
Take a look and tell me what you think.

kamil-kaczmarek commented:
ok, @jakubczakon let me have a look tomorrow morning

kamil-kaczmarek commented on integrations-and-supported-tools/kedro/scripts/kedro_neptune_advanced/conf/base/catalog.yml:
```diff
@@ -0,0 +1,64 @@
+#
+# An example data set definition can look as follows:
+#
+#bikes:
+#  type: pandas.CSVDataSet
+#  filepath: "data/01_raw/bikes.csv"
+#
+#weather:
+#  type: spark.SparkDataSet
+#  filepath: s3a://your_bucket/data/01_raw/weather*
+#  file_format: csv
+#  credentials: dev_s3
+#  load_args:
+#    header: True
+#    inferSchema: True
+#  save_args:
+#    sep: '|'
+#    header: True
+#
+#scooters:
+#  type: pandas.SQLTableDataSet
+#  credentials: scooters_credentials
+#  table_name: scooters
+#  load_args:
+#    index_col: ['name']
+#    columns: ['name', 'gear']
+#  save_args:
+#    if_exists: 'replace'
+#    # if_exists: 'fail'
+#    # if_exists: 'append'
+#
+# The Data Catalog supports being able to reference the same file using two different DataSet implementations
+# (transcoding), templating and a way to reuse arguments that are frequently repeated. See more here:
+# https://kedro.readthedocs.io/en/stable/05_data/01_data_catalog.html
+#
+# This is a data set used by the "Hello World" example pipeline provided with the project
+# template. Please feel free to remove it once you remove the example pipeline.
```
I would drop all unnecessary things from this file. It's a bit misleading, as I do not know what is relevant to the example and what is not.

kamil-kaczmarek commented on integrations-and-supported-tools/kedro/scripts/kedro_neptune_advanced/conf/base/catalog_neptune.yml:
```diff
@@ -0,0 +1,14 @@
+# You can log files to Neptune via NeptuneFileDataSet
+#
+# example_artifact:
+#   type: kedro_neptune.NeptuneFileDataSet
+#   filepath: data/06_models/clf_model.pkl
+#
+# If you want to log existing Kedro Dataset to Neptune add @neptune to the DataSet name
+#
+# example_iris_data@neptune:
+#   type: kedro_neptune.NeptuneFileDataSet
+#   filepath: data/01_raw/iris.csv
+#
+# You can use kedro_neptune.NeptuneFileDataSet in any catalog including conf/base/catalog.yml
+#
```
Seems like the entire config is commented out. I wonder if we actually need it?

kamil-kaczmarek commented on integrations-and-supported-tools/kedro/scripts/kedro_neptune_advanced/src/kedro_neptune_advanced/pipelines/data_science/nodes.py:
```diff
@@ -0,0 +1,116 @@
+
+
+def evaluate_models(predictions: dict, test_y: pd.DataFrame,
+                    neptune_run: neptune.run.Handler):
```
Not sure, but it probably should be 
`neptune_run: neptune.run.Run` -> it's neptune Run object that you use for logging.

kamil-kaczmarek commented on integrations-and-supported-tools/kedro/scripts/kedro_neptune_advanced/src/kedro_neptune_advanced/pipelines/data_science/nodes.py:
```diff
@@ -0,0 +1,116 @@
+
+
+def ensemble_models(predictions: dict, test_y: pd.DataFrame,
+                    neptune_run: neptune.run.Handler) -> np.ndarray:
```
Not sure, but it probably should be
`neptune_run: neptune.run.Run` -> it's neptune Run object that you use for logging.

jakubczakon replied:
it is actualy a handler afaik,

kamil-kaczmarek commented on integrations-and-supported-tools/kedro/scripts/kedro_neptune_advanced/src/kedro_neptune_advanced/pipelines/data_science/pipeline.py:
```diff
@@ -0,0 +1,74 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Example code for the nodes in the example pipeline. This code is meant
+just for illustrating basic Kedro features.
+
+Delete this when you start working on your own Kedro project.
+"""
```
maybe dropping this

jakubczakon replied:
mhm we could -> but as the example starts with the kedro starter that has it and builds on top of it, I think I'd leave as is.

kamil-kaczmarek reviewed (approved):
left few comment, nothing is blocking
